### ISSUE TITLE
backend serve cannot run on windows

### ISSUE THREAD

**Issue by miaomiao1992** [rust-lang/rust-playground#913]
it seems only linux can deploy the server.

```
error[E0433]: failed to resolve: could not find `unix` in `os`
 --> src\sandbox.rs:8:9
  |
8 |     os::unix::fs::PermissionsExt,
  |         ^^^^ could not find `unix` in `os`

error[E0433]: failed to resolve: could not find `unix` in `os`
   --> src\sandbox.rs:624:18
    |
624 |     use std::os::unix::process::ExitStatusExt;
    |                  ^^^^ could not find `unix` in `os`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `ui` due to 2 previous errors

```

**Comment by shepmaster** [comment]
Yes, the current implementation expects to communicate with the Docker instance in part by writing files to disk from the server and then reading them inside the container (and vice-versa for the result).

Work is progressing on a re-implementation where we will likely _only_ communicate to the Docker container via stdin/out, which should magically fix this problem.

In the short term, try commenting that code out and seeing how running it works. It may not be needed at all on Windows, or it may need some alternative. Happy to review a PR with the results.

**Comment by shepmaster** [comment]
The `ExitStatusExt` is trickier. The container definitely runs Linux commands and we get the Linux exit code. We smuggle that through the Docker layer and re-attach it to propagate upwards. It's possible we may only need a simple success/failure indicator instead of the full-blown exit code though.

Ditto for experimentation and future PR.
